moduleframework/common.py

```diff
@@ -252,36 +201,25 @@ def get_odcs_auth():
     :envvar MTF_ODCS: yes or token
     :return:
     """
-    odcstoken = os.environ.get('MTF_ODCS')
+    odcstoken = get_odcs_envvar()
 
     # in case you dont have token enabled, try to ask for openidc via web browser
-    if odcstoken in TRUE_VALUES_DICT:
+    if is_true(odcstoken):
+        if conf.get("openidc").get("token"):
+            # use value defined in config file if defined
+            return conf["openidc"]["token"]
         # to not have hard dependency on openidc (use just when using ODCS without defined token)
         import openidc_client
-        id_provider = 'https://id.fedoraproject.org/openidc/'
         # Get the auth token using the OpenID client.
-        oidc = openidc_client.OpenIDCClient(
-            'odcs',
-            id_provider,
-            {'Token': 'Token', 'Authorization': 'Authorization'},
-            'odcs-authorizer',
-            'notsecret',
-        )
-
-        scopes = [
-            'openid',
-            'https://id.fedoraproject.org/scope/groups',
-            'https://pagure.io/odcs/new-compose',
-            'https://pagure.io/odcs/renew-compose',
-            'https://pagure.io/odcs/delete-compose',
-        ]
+        oidc = openidc_client.OpenIDCClient(*conf["openidc"]["auth"])
```

Comment:
@jscotka Check if ["openidc"]["auth"] exists.
what about if oidc is None. Where do you catch traceback?